csharp/src/Google.Protobuf/JsonFormatter.cs

```diff
@@ -516,6 +541,12 @@ private void WriteFieldMask(StringBuilder builder, IMessage value)
 
         private void WriteAny(StringBuilder builder, IMessage value)
         {
+            if (ReferenceEquals(this, diagnosticFormatter))
+            {
+                WriteDiagnosticOnlyAny(builder, value);
+                return;
+            }
+
             string typeUrl = (string) value.Descriptor.Fields[Any.TypeUrlFieldNumber].Accessor.GetValue(value);
             ByteString data = (ByteString) value.Descriptor.Fields[Any.ValueFieldNumber].Accessor.GetValue(value);
             string typeName = GetTypeName(typeUrl);
```

Comment:
Do we have any or if it is possible to add tests for Any message when use TypeRegistry?